tests/lib/snaps/test-snapd-content-plug-to-classic-slot/meta/snap.yaml

```diff
@@ -0,0 +1,12 @@
+name: test-snapd-content-plug-to-classic-slot
+version: 1.0
+apps:
+    content-plug:
+        command: bin/content-plug
+        plugs: [shared-content-plug]
+plugs:
+    shared-content-plug:
+        interface: content
+        target: import
+        content: myclib
+        default-provider: test-snapd-classic-content-slot
```

Comment:
hmm I think this bit brings up a good policy question - do we allow classic snaps to be installed via default-providers ?

I don't think it should, imagine the use-case where I as a security conscious user look at a snap and decide to install it, it is a strict snap, so I use `snap install ...` without the `--classic` flag since it is not a classic snap and I am not consenting to install a classic snap - should snapd install a classic snap without my explicit permission given that the classic snap could contain any sort of unconfined code executed on my system. I think this breaks the trust of the strict snap user. I think we can allow content interface slots from snaps, but the user has to manually install those classic content providing snaps. If this user experience isn't good enough, then to use default-provider you need to have the content providing snap be strict.

This is further complicated by the fact that we don't expose to the user any sort of way to inspect what other snaps are pulled in when you install a given snap, except perhaps by manually downloading and unpacking the snap in question.

Reply:
I suppose this will not work because we don't pass the right flags for this when we install default-providers, I suppose that is good. But yes, thinking further I don't think we can allow for this